### DISCUSSION TITLE
Flask pymodbus  implementation

### DISCUSSION THREAD

**Discussion by nguefeu** [pymodbus-dev/pymodbus#1238] [Help] [answered]
ello,

I am student and I working on integration of Raspberry Pi as Modbus client for industrial TCP connection.
I got some problem to evaluate connection to Modbus TCP server (sensor) and read register from a database.

 Here is my test connection function

import json

from pymodbus.client import ModbusTcpClient

# clientStatusController BEGIN
def clientStatusController(request_data=False):
  modbusClient=ModbusTcpClient("localhost", port=5020).is_socket_open()

  if not request_data:
    return json.dumps(modbusClient)

  open_client=False
  #print(request_data)
  # LOGGING CANCELED logging.debug(request_data)
  try:
    open_client = json.loads(request_data)["open_client"]
  except Exception as exception:
    responseObject={"host":modbusClient.host(),"port":modbusClient.port(),"message":"Bad Parameter"}
    return json.dumps(responseObject)
`

**Comment by janiversen** [comment, marked as answer]
Please read the documentation on how to use pymodbus or use one of the examples (like https://github.com/riptideio/pymodbus#example-code).

Creating an object of ModbusTcpClient does not make a connection.

Secondly you cannot use json directly with the modbus protocol, you need to call a modbus service, like e.g. ReadHoldingRegister()
